packages/itwinui-react/scripts/postBuild_Types.mjs

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { exec } from 'node:child_process';
+
+// Since tsc does not include the react-table-types.d.ts file in the cjs and esm builds,
+// we need to copy it manually.
+['cjs', 'esm'].forEach((buildType) => {
+  exec(
+    `copyfiles -u 1 src/core/Table/types/react-table-types.d.ts ${buildType}`,
```

Comment:
you don't need this script. if you rename the file extension from `d.ts`, then it should get compiled correctly

Reply:
Rename it to which file extension? `.ts` causes other errors. E.g. something along the lines of, "you are describing a function without any implementation".

Reply:
yes, `.ts`. could you post the exact error?

Reply:
<!--![image](https://github.com/iTwin/iTwinUI/assets/45748283/7647fb44-2a44-45a1-ad28-bf5fc722d3de) -->
![image](https://github.com/iTwin/iTwinUI/assets/45748283/5d829512-eb8c-40a3-a6e7-74e55b250584)

Reply:
should be fixable using `declare` i think.

```ts
export declare const useAbsoluteLayout: (<
  D extends Record<string, unknown> = {},
>(
  hooks: Hooks<D>,
) => void) & { pluginName: 'useAbsoluteLayout' };
```

also it looks like these errors are only for the plugin hooks, which do not matter for public types. so we could even remove these

Reply:
I noticed that adding `declare` to the function and removing the `export namespace` solves the errors.

```ts
export declare function useAbsoluteLayout<
  D extends Record<string, unknown> = {},
>(hooks: Hooks<D>): void;

// export namespace useAbsoluteLayout {
//   const pluginName = 'useAbsoluteLayout';
// }
```

To confirm, what is the significance of exporting namespaces like `useAbsoluteLayout`, `useColumnOrder`, etc.? Would just exporting the functions be enough?

Also, apart from these plugin hooks that give errors, there are also some `const`s ([diff](https://github.com/iTwin/iTwinUI/commit/f7140d13e40d511ac8ff77e6150e93f270effbab#diff-fa89432ee3e85390e059a0f5f01419a9bb62c8a0f306aa426ae26fbe1f825ee3L1218-R1228)) and `function`s ([diff](https://github.com/iTwin/iTwinUI/commit/f7140d13e40d511ac8ff77e6150e93f270effbab#diff-fa89432ee3e85390e059a0f5f01419a9bb62c8a0f306aa426ae26fbe1f825ee3R1250-R1297)) that were undefined. So I used `declare` for them too.

I did the above mentioned points in a temporary commit https://github.com/iTwin/iTwinUI/pull/1514/commits/f7140d13e40d511ac8ff77e6150e93f270effbab. I also added a few temporary eslint ignore comments at the top of the file. This is just so we can focus on this error first before moving to other errors.

Also, I am wondering, is it semantically okay to have unimplemented functions in `.ts` files using `declare`? I feel `.d.ts` files are better suited for only have types.

Reply:
Ohh, I see, sounds good.

> you can check the built output to be sure that there is no unexpected output in `.js`.

The built react-table-types.js file for the esm and cjs folders are nearly empty.

ESM:

![image](https://github.com/iTwin/iTwinUI/assets/45748283/0ff207cf-a88d-4d97-b0b9-227e333a0a2c)

CJS:

![image](https://github.com/iTwin/iTwinUI/assets/45748283/4ad0fef6-6070-4403-abe0-516483648460)

Reply:
> > you can check the built output to be sure that there is no unexpected output in `.js`.
> 
> The built react-table-types.js file for the esm and cjs folders are nearly empty.

yup, that's what i mean by "no unexpected output". we expect it to be empty.

Reply:
so this postBuild_Types file can be removed now?

Reply:
Yes, removed (https://github.com/iTwin/iTwinUI/pull/1514/commits/230b73477716175e772c7206bf0bf4170d4d1536)

Reply:
the `copyfiles` dependency can also be removed

Reply:
Removed (https://github.com/iTwin/iTwinUI/pull/1514/files/6ccf4500a988aeab854a7aa1e221a440033a2de0..e9c27d798ba33b3d7c50d37945e9438d38f93080)